### ISSUE TITLE
Add getGroupName method to ApiListingContext

### ISSUE THREAD

**Issue by walec51** [springfox/springfox#2641]
All other contexts have it. I was surprised that ApiListingContext has not and I could not implement my plugin that required to know the group name.
ApiListingContext has `.getResourceGroup().getGroupName()` but this is something different.

**Comment by walec51** [comment]
I'm writing a `ApiListingBuilderPlugin` and I see no `.getDocumentationContext()` method in `ApiListingContext`. How could I get the document context from `ApiListingBuilderPlugin::apply` ?

**Comment by dilipkrish** [comment]
You're right, sorry! I was thinking ApiListingScannerContext. Why do you need it there again?

**Comment by walec51** [comment]
I just have some functionality that has to know the group name. It seems its available in all other contexts except this one.

In my use case I have it translations based on group name. In my plugins I check if a group name ends with a suffix like "-locale-pl" or "-locale-de". If it does I extract metadata for that specified language if its available. In the case of ApiListingBuilderPlugin I want to provide a locale dependent description of the listing.

If there was a way to pass custom properties to a docket that would be available for every plugin I could build a even cleaner solution instead of extracting data from group name. However I would still have a group per module and per language in my case and still it seems inconsistent that most plugins can assess the group name from their context but ApiListingBuilderPlugin cannot.
